File: Editor/assets/Scripts/GameState/InitHub.cs
using Wiwa;

namespace WiwaApp
{
    class InitHub : Behaviour
    {
        private bool pendingToLoad = true;
        void Update()
        {
            if (pendingToLoad)
            {
                GameState.InitHub(m_Scene);
                pendingToLoad = false;
            }
        }
    }
}

File: Editor/game_assembly/src/systems/triggers/InitHub.cpp
#include "InitHub.h"

Wiwa::InitHubTrigger::InitHubTrigger()
{
}

Wiwa::InitHubTrigger::~InitHubTrigger()
{
}

void Wiwa::InitHubTrigger::OnAwake()
{
	GameStateManager::InitHub();
}
